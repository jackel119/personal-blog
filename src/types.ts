import { MDXRemoteSerializeResult } from "next-mdx-remote";

export interface FrontMatter {
  title: string;
  date: string;
  abstract: string;
}

export interface Post {
  id: string;
  info: FrontMatter;
  mdx: MDXRemoteSerializeResult;
}
