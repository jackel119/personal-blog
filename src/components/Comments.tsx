import { useEffect, useRef } from "react";

import { useDarkMode } from "hooks/useDarkMode";

const SCRIPT_ELEM_ID = "utterancesScript";
export function Comments() {

  const { dark } = useDarkMode();
  const ref = useRef<HTMLDivElement>(null);

  const theme = dark ? "dark-blue" : "github-light";

  useEffect(() => {

    if (!ref.current) return () => undefined;

    const scriptEl = document.createElement("script");
    scriptEl.id = SCRIPT_ELEM_ID;
    scriptEl.setAttribute("src", "https://utteranc.es/client.js");
    scriptEl.setAttribute("crossorigin", "anonymous");
    scriptEl.setAttribute("async", "true");
    scriptEl.setAttribute("repo", "jackpordi/personal-blog");
    scriptEl.setAttribute("issue-term", "pathname");
    scriptEl.setAttribute("theme", theme);

    ref.current!.appendChild(scriptEl);
    return () => {
      document.getElementById(SCRIPT_ELEM_ID)?.remove();
      document.getElementsByClassName("utterances")[0]?.remove();
    };
  }, []);

  useEffect(() => {
    if (document.querySelector(".utterances-frame")) {
      const iframe = document.querySelector<HTMLIFrameElement>(".utterances-frame");

      if (!iframe) {
        return;
      }
      iframe?.contentWindow?.postMessage({
        type: "set-theme",
        theme,
      }, "https://utteranc.es");
    }
  }, [ dark ]);

  return (
    <div style={{ width: "100%" }} id="comments">
      <section ref={ref}/>
    </div>
  );
}
