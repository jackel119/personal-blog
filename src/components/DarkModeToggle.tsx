import { FaMoon, FaSun } from "react-icons/fa";

import { useDarkMode } from "hooks/useDarkMode";

export const DarkModeToggle = () => {

  const { dark, toggle } = useDarkMode();

  return (
    <div
      className="cursor-pointer mr-6"
      onClick={(e) => {
        e.preventDefault();
        toggle();
      }}
    >
      { dark
        ? <FaMoon
          className="text-white"
          size={24}/>
        : <FaSun
          color="orange"
          size={24}/>
      }
    </div>
  );
};
