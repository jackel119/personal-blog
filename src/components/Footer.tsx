
export function Footer() {
  return (
    <footer
    className="items-center justify-center flex-grow flex py-4 text-gray-600"
    >
      &copy; 2021 Jack Pordi, All rights reserved
    </footer>
  );
}
