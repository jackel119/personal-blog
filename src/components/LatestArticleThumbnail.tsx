import Image from "next/image";
import { FC } from "react";
import Link from "next/link";

import { useDisplayDate } from "hooks/useDisplayDate";
import { PostDetails } from "types";

export const LatestArticleThumbnail: FC<PostDetails> = (post) => {
  const { age } = useDisplayDate(post.info.date);
  return (
    <div className="col-span-4 group">
      <Link href={`/posts/${post.id}`}>
        <Image
          src={post.info.image!}
          alt={post.info.title}
          height={150}
          width={200}
          style={{
            height: 150,
            width: "100%",
            objectFit: "cover",
          }} />
        <h5 className="fade-colors text-xl text-black dark:text-gray-200 dark:group-hover:text-cyan-400 group-hover:text-cyan-500 font-semibold">{ post.info.title }</h5>

        <div className="text-gray-400 dark:text-gray-200 flex flex-row">{ post.readingTime.toFixed(0)} min read | {age}</div>
      </Link>
    </div>
  );
};
