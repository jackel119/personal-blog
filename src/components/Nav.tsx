import Link from "next/link";
import { FaEnvelope, FaGithub } from "react-icons/fa";
import { FunctionComponent } from "react";

export const Nav: FunctionComponent = () => (
  <nav className="inset-x-0 flex flex-1 items-center justify-between flex-wrap p-6 py-22 text-gray-500">
    <a
      href="/"
      className="flex items-center flex-shrink-0 mr-6"
    >
      <span className="text-gray-700 text-3xl md:text-3xl tracking-tight">
        Jack
        {" "}
        <span className="text-blue-600 font-bold">Pordi</span>
      </span>
    </a>
    <div className="block lg:hidden">
      <button className="flex items-center px-3 py-2 border rounded text-black border-teal-400 hover:border-white">
        <svg className="fill-current h-3 w-3" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
          <title>Menu</title>
          <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z" />
        </svg>
      </button>
    </div>
    <div className="w-full lg:flex-grow hidden lg:flex lg:items-center lg:w-auto">
      <div className="text-xl lg:flex-grow lg:items-center lg:flex justify-end">
        <Link href="/" passHref>
          <div className="block mt-4 lg:inline-block lg:mt-0 cursor-pointer hover:text-blue-600 hover:font-semibold mr-6 transition transform hover:scale-115">
            Home
          </div>
        </Link>
        <a
          className="mr-6 transition transform hover:scale-115 hover:text-blue-600"
          href="mailto: contact@jackpordi.com"
        >
          <FaEnvelope
            size={28}
          />
        </a>
        <a
          className="mr-6 transition transform hover:scale-115 hover:text-blue-600"
          href="https://github.com/jackpordi"
        >
          <FaGithub
            size={28}
          />
        </a>
      </div>
    </div>
  </nav>
);
