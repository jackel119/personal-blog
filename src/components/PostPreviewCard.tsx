import { FrontMatter } from "types";

interface Props {
  post: FrontMatter;
}

export function PostPreviewCard({ post }: Props) {
  return (
    <a href="https://nextjs.org/docs" className="transform transitionmargin-2 text-left group my-4 max-w-2xl">
      <h2 className="text-2xl group-hover:text-blue-600 font-semibold">{post.title}</h2>
      <p className="text-gray-700 text-lg mt-2 group-hover:text-black ">
      { post.abstract }
      </p>
      <p className="text-black mt-4">
        <span className="group-hover:text-blue-600">Read More &rarr;</span>
      </p>
    </a>
  );
}
