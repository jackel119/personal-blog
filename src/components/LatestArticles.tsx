import { FC } from "react";

import { PostDetails } from "types";

import { LatestArticleThumbnail } from "./LatestArticleThumbnail";

interface Props {
  latest: PostDetails[];
}

export const LatestArticles: FC<Props> = ({ latest }) => (
  <div className="mb-2 mt-4 hidden md:block">
    <h4 className="text-3xl font-semibold dark:text-gray-200">Latest Posts</h4>
    <div className="grid grid-cols-12 gap-4 my-1">
      { latest.map((post) => <LatestArticleThumbnail key={post.id} {...post} />)}
    </div>
  </div>
);
