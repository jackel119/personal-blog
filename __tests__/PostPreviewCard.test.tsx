import { render, screen } from '@testing-library/react'
import moment from 'moment';

import { FrontMatter } from 'types';
import { PostPreviewCard } from "components/PostPreviewCard";

describe('PostPreviewCard', () => {

  const POST_ID = "test-post-id";
  const POST: FrontMatter = {
    title: "Example Post Title",
    date: moment(0).format(),
    abstract: "This is a sample article abstract",
    image:  "https://image-url.com",
  };

  it("Renders the post title", () => {
    render(<PostPreviewCard
      id={POST_ID}
      post={POST}
    />);

    const title = screen.getByRole('heading', {
      name: POST.title,
    })

    expect(title).toBeInTheDocument()
  });

  it("Renders the post abstract", () => {
    render(<PostPreviewCard
      id={POST_ID}
      post={POST}
    />);

    const abstract = screen.getByText(POST.abstract);
    expect(abstract).toBeInTheDocument()
  });

  it("Renders the post image", () => {
    render(<PostPreviewCard
      id={POST_ID}
      post={POST}
    />);

    const img = screen.getByAltText(POST.image!);
    expect(img).toBeInTheDocument()
  });

  it("Matches the snapshot", () => {
    const { container } = render(<PostPreviewCard
      id={POST_ID}
      post={POST}
    />);
    expect(container).toMatchSnapshot()
  })

})
