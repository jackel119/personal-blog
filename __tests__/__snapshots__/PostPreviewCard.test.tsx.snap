// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPreviewCard Matches the snapshot 1`] = `
<div>
  <a
    href="/posts/test-post-id"
  >
    <div
      class="md:px-4 px-3 py-1 md:py-4 flex-row flex w-full flex-1 transform transition margin-2 text-left group my-2 max-w-2xl cursor-pointer rounded-sm"
    >
      <div
        class="mr-4 justify-center"
      >
        <div
          class="rounded-sm hidden md:flex"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27180%27%20height=%27150%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="https://image-url.com"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="rounded-sm md:hidden flex"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27120%27%20height=%27100%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="https://image-url.com"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </span>
        </div>
      </div>
      <div
        class="flex-1 flex flex-col md:max-h-40"
      >
        <h2
          class="fade-colors text-xl text-black dark:text-gray-200 dark:group-hover:text-cyan-400 group-hover:text-cyan-500 font-semibold flex-1 md:flex-none"
        >
          Example Post Title
        </h2>
        <p
          class="text-gray-700 text-md group-hover:text-black dark:text-gray-300 dark:group-hover:text-white hidden md:flex"
        >
          This is a sample article abstract
        </p>
         
        <div
          class="flex flex-col-reverse md:flex-row text-black md:mt-1 justify-between w-full"
        >
          <div
            class="fade-colors group-hover:text-cyan-500 dark:text-gray-200 dark:group-hover:text-cyan-400"
          >
            Read More â†’
          </div>
          <div
            class="text-gray-400 dark:text-gray-200 flex flex-row"
          >
            3
             min read | 
            52 years ago
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
